src/toolkits/coreml_export/neural_net_models_exporter.cpp

```diff
@@ -185,20 +186,30 @@ std::shared_ptr<MLModelWrapper> export_object_detector_model(
   model_nn->mutable_neuralnetwork()->MergeFrom(nn_spec.get_coreml_spec());
   ASSERT_GT(model_nn->neuralnetwork().layers_size(), 1);
 
+  // Set the name of preprocessing layer to input_name as well.
+  // Since in the test case somehow the model doesn't have a preprocessing
+  // layer, so we need to check the size first...
+  if (model_nn->mutable_neuralnetwork()->mutable_preprocessing()->size() > 0) {
```

Comment:
Do not use `mutable` accessors here, since they will create these fields if they didn't exist before. Just replace e.g. `mutable_neuralnetwork()->` with `neuralnetwork().`

Reply:
This is a really good point. Thanks!